webcomponents/tool/src/main/frontend/js/calendar/sakai-calendar.js

```diff
@@ -171,9 +171,11 @@ export class SakaiCalendar extends LionCalendar {
 
   render() {
 
+    const daysMessage = `${this.i18n.days_message.replace("{}", this.days)}`;
```

Comment:
You shouldn't be doing this as it will execute this replace on every render. If you keep it in the html template, lit will know if it needs to update it. Otherwise it won't be executed. So, what you should do in this case is add a shouldUpdate function which checks for this.i18n existing. Have a look at another component - it happens in lots of places. On the other hand, you could requestUpdate in the promise handler where the i18n is initially loaded.

Reply:
Thanks for pointing out this important optimization.